client/normal_new.go

```diff
@@ -0,0 +1,465 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package client
+
+import (
+	"time"
+
+	"github.com/sourcenetwork/immutable"
+)
+
+// NewNormalValue creates a new NormalValue from the given value.
+// It will normalize all known types that can be converted to normal ones.
+// For example, is the given type is `[]int32`, it will be converted to `[]int64`.
+// If the given value is of type `[]any` is will to through every element and try to convert it
+// minimal common type and normalizes it.
```

Comment:
question: Is `minmal` the right word here? There might be something missing to make it make sense.